### ISSUE TITLE
`Bus::chain` Skipping jobs with readonly non-scalar properties

### ISSUE THREAD

**Issue by shaffe-fr** [laravel/framework#51877]
### Laravel Version

10.x and 11.x

### PHP Version

8.1, 8.2 and 8.3

## Description of the Issue
When a `Bus::chain` contains a job with a readonly promoted non-scalar property, only the first job or batch in the chain is executed. The subsequent jobs or batches appear to be discarded. Additionally, even if the chain is dispatched on a non-default queue, the jobs still run on the default queue.

No exceptions were found in the logs.

## Environment
Laravel versions: 10.x and 11.x
PHP versions: 8.1, 8.2, and 8.3

## Evidence
Failing test PR : #51878 
A repository that reproduces the problem can be found here: https://github.com/shaffe-fr/framework/commit/f1b3cad6d6e3a78d108f25d751d144049e070626

Here is a screenshot illustrating the issue: 
![image](https://github.com/laravel/framework/assets/3834222/577bfcec-eeae-44a5-b930-f534c9cd380c)

### Steps To Reproduce

1. Create a `Bus::chain` with at least one job having a readonly promoted non-scalar property.
2. Dispatch the chain.
3. Observe that only the first job or batch is executed, and subsequent jobs/batches are not processed.
4. Note that jobs run on the default queue, regardless of the specified queue.

**Comment by driesvints** [comment]
Thanks @shaffe-fr. It's not by consequence that this is related to the enum usages in your PR?

**Comment by shaffe-fr** [comment]
Hi @driesvints, indeed, I had the same issue with classes. 
I added another failing test case with a simple class in the PR #51878.

**Comment by juse-less** [comment]
The issue I have with chained batches is that they stop executing whenever I have a job that uses a class type for its properties.
As soon as the next job in a chained batch has a class property, the chain is removed from the queue and marked as completed.

I've tried many combinations of visibility, readonly, serialising and unserialising the classes myself 
Nothing seems to work, other than removing the properties.

All the jobs/batches in the chain are correctly displayed in the payload in both the Redis and database drivers.

**Comment by rust17** [comment]
@juse-less Could you provide some example codes? I can't reproduce it.

**Comment by juse-less** [comment]
@rust17 Hmm.
I might have a different issue entirely, but the behaviour sounds like the same as this issue.

I'll try your fix today, just to be sure I'm not missing something.
I'll then create a minimal reproducer.

**Comment by shaffe-fr** [comment]
I tried your fix with the code in the PR #51878 but it does not seems to work.

**Comment by rust17** [comment]
@shaffe-fr Hmm. This looks strange. I tried it and it works for me. You can see the results [here](https://github.com/rust17/framework/actions). Could you consider accepting this [PR](https://github.com/shaffe-fr/framework/pull/1)?

**Comment by shaffe-fr** [comment]
I tried a quick and dirty test: I added the if statement in the Native.php file modification in the vendor.

I'll try it directly in my project using your version of laravel/serializable-closure.

**Comment by shaffe-fr** [comment]
I confirm that I don't have the problem in my project with this fix!

**Comment by juse-less** [comment]
Same here. Seems like it actually fixes the problem.

I tested on the state I left things in a couple weeks ago, but definitely works now.
I first tested without installing @rust17 branch, and it still failed.
After installing the branch, it now goes through a chain of 10 batches, most of which use class properties- and arguments.

I'm gonna reset all my changes I did while debugging, but I'm confident this does solve it altogether.
